vtkext/private/module/vtkF3DMetaImporter.cxx

```diff
@@ -470,48 +497,52 @@ bool vtkF3DMetaImporter::UpdateAtTimeValue(double timeValue)
     importerPair.Importer->UpdateTimeStep(timeValue);
 #endif
   }
+  this->Pimpl->UpdateTime.Modified();
   return ret;
 }
 
 //----------------------------------------------------------------------------
 void vtkF3DMetaImporter::UpdateInfoForColoring()
 {
-  for (const auto& importerPair : this->Pimpl->Importers)
+  if (this->Pimpl->UpdateTime.GetMTime() > this->Pimpl->ColoringInfoTime.GetMTime())
   {
+    for (const auto& importerPair : this->Pimpl->Importers)
+    {
 #if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20240707)
-    vtkActorCollection* actorCollection = importerPair.Importer->GetImportedActors();
+      vtkActorCollection* actorCollection = importerPair.Importer->GetImportedActors();
 #else
-    vtkActorCollection* actorCollection =
-      this->Pimpl->ActorsForImporterMap.at(importerPair.Importer).Get();
+      vtkActorCollection* actorCollection =
+        this->Pimpl->ActorsForImporterMap.at(importerPair.Importer).Get();
 #endif
-    vtkCollectionSimpleIterator ait;
-    actorCollection->InitTraversal(ait);
-    while (auto* actor = actorCollection->GetNextActor(ait))
-    {
-      vtkPolyDataMapper* pdMapper = vtkPolyDataMapper::SafeDownCast(actor->GetMapper());
-      assert(pdMapper);
-
-      // Update coloring vectors, with a dedicated logic for generic importer
-      vtkDataSet* datasetForColoring = pdMapper->GetInput();
-      vtkF3DGenericImporter* genericImporter = vtkF3DGenericImporter::SafeDownCast(importerPair.Importer);
-      if (genericImporter)
+      vtkCollectionSimpleIterator ait;
+      actorCollection->InitTraversal(ait);
+      while (auto* actor = actorCollection->GetNextActor(ait))
       {
-        // TODO This will be improved with proper composite support
-        // Currently generic importer always has a single actor
-        if (genericImporter->GetImportedImage())
-        {
-          datasetForColoring = genericImporter->GetImportedImage();
-        }
-        else if (genericImporter->GetImportedPoints())
+        vtkPolyDataMapper* pdMapper = vtkPolyDataMapper::SafeDownCast(actor->GetMapper());
+        assert(pdMapper);
+
+        // Update coloring vectors, with a dedicated logic for generic importer
+        vtkDataSet* datasetForColoring = pdMapper->GetInput();
+        vtkF3DGenericImporter* genericImporter = vtkF3DGenericImporter::SafeDownCast(importerPair.Importer);
+        if (genericImporter)
         {
-          datasetForColoring = genericImporter->GetImportedPoints();
+          // TODO This will be improved with proper composite support
```

Comment:
Avoid the `TODO` keyword if you don't plan to do it in this PR.

Reply:
I'm just moving this comment around, but I can write XXX instead if you prefer.